activesupport/lib/active_support/cache/strategy/local_cache.rb

```diff
@@ -64,8 +64,9 @@ def read_multi_entries(keys, **options)
             values
           end
 
-          def write_entry(key, value, **options)
-            @data[key] = value
+          def write_entry(key, entry, **options)
+            entry.dup_value!
```

Comment:
will this do deep duping?

Reply:
It does a Marshal.load https://github.com/rails/rails/blob/966d3a7bf2c08c5c20213d29686cbb409a07734a/activesupport/lib/active_support/cache.rb#L813